Add table of contents to the README

mcdaniel67 opened the pull request:
Addresses https://github.com/pixielabs/cavy/issues/48

Added anchors for headings and table of contents to the main readme.

See readme here:
https://github.com/mcdaniel67/cavy/tree/feature__readme_table_of_contents

AbigailMcP commented:
Hey @mcdaniel67, thanks for this!

I think you should be able to ditch the \<a> tags though, with punctuation marks dropped for hyphens e.g.

`[Cavy's Components](#cavy-s-components)` will automatically link to header `#Cavy's components`

mcdaniel67 commented:
Fixed, I was hesitant to do it this way at first because I wasn't sure how links like this work in the case where we are linking to something with punctuation. So my TIL for today is to just ignore it. 

Also just to share a bit about what I learned, the Cavy's components part is actually like `- [Cavy's Components](#cavys-components)` you totally ignore the `'`

Finally, I updated the bullets to use the same syntax as the "Contribution" section after I saw what it looks like when starting to add the contribution code of conduct.

jalada commented:
Thanks @mcdaniel67 !
